include/git2/worktree.h

```diff
@@ -78,10 +78,11 @@ typedef struct git_worktree_add_options {
 	unsigned int version;
 
 	int lock; /**< lock newly created worktree */
+	git_reference *ref; /**< reference to use for the new worktree HEAD */
 } git_worktree_add_options;
 
 #define GIT_WORKTREE_ADD_OPTIONS_VERSION 1
```

Comment:
When/how does the version get bumped?

Reply:
The option versions shall get bumped whenever the structure changes, but this will only happen as soon as we have our first stable (1.0) release.

Reply:
I have a general question about changing this struct/the version and breaking changes, really for my own education.

In libgit2sharp the shape of the _options structs is effectively fixed, e.g. for git_blame_options
see https://github.com/libgit2/libgit2sharp/blob/3f9b415fa1edfc31ce1ec2b4b3d18441c34adfff/LibGit2Sharp/Core/GitBlame.cs#L44


    [StructLayout(LayoutKind.Sequential)]
    internal class git_blame_options
    {
        public uint version = 1;
        public GitBlameOptionFlags flags;

        public UInt16 min_match_characters;
        public git_oid newest_commit;
        public git_oid oldest_commit;
        public UIntPtr min_line;
        public UIntPtr max_line;
    }

vs in libgit2 https://github.com/libgit2/libgit2/blob/89c332e41b12a72d89de40d63bc568c56a2c336a/include/git2/blame.h#L70


    [StructLayout(LayoutKind.Sequential)]
    internal class git_blame_options
    {
        public uint version = 1;
        public GitBlameOptionFlags flags;

        public UInt16 min_match_characters;
        public git_oid newest_commit;
        public git_oid oldest_commit;
        public UIntPtr min_line;
        public UIntPtr max_line;
    }

So the libgit2sharp code would need reworking if a new field was added to the struct in libgit2.

So my question is why is it ok to refactor the worktree options struct, without bumping the version, in libgit2 but not to add params to a method definition?

Why does adding a param to the method break the API but adding a field to the struct not break it? 

Hope that makes sense. :)

Reply:
You're very much right in that _both_ are breaking API changes.  The way that we extend things now is (generally) through the options interface instead of params.  In the future it will always be through the options interface plus incrementing the version number.

We haven't been very responsible here about stabilizing.  We should do that soon.